packages/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -1,26 +1,245 @@
+    borderRadius: theme.global.borderRadius.circular,
+  }),
+
+  disabled: theme => ({
+    borderWidth: theme.global.strokeWidth.thin,
+    borderColor: theme.alias.color.neutral.neutralStrokeDisabled,
+  }),
+
+  unchecked: theme => ({
+    borderColor: theme.alias.color.neutral.neutralStrokeAccessible,
+    borderWidth: theme.global.strokeWidth.thin,
+    ':hover': {
+      borderColor: theme.alias.color.neutral.neutralStrokeAccessibleHover,
+    },
+    ':active': {
+      borderColor: theme.alias.color.neutral.neutralStrokeAccessiblePressed,
+    },
+  }),
+
+  checked: theme => ({
+    backgroundColor: theme.alias.color.neutral.compoundBrandBackground,
+    borderWidth: 0,
+    ':hover': {
+      backgroundColor: theme.alias.color.neutral.compoundBrandBackgroundHover,
+    },
+    ':active': {
+      backgroundColor: theme.alias.color.neutral.compoundBrandBackgroundPressed,
+    },
+  }),
+
+  mixed: theme => ({
+    borderColor: theme.alias.color.neutral.compoundBrandStroke,
+    borderWidth: theme.global.strokeWidth.thin,
+    ':hover': {
+      borderColor: theme.alias.color.neutral.compoundBrandStrokeHover,
+    },
+    ':active': {
+      borderColor: theme.alias.color.neutral.compoundBrandStrokePressed,
+    },
+  }),
+});
+
+const useIndicatorStyles = makeStyles({
+  indicator: {
+    position: 'absolute',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+
+  // TODO: Remove fontSize once checkbox uses react-icons
+  medium: {
+    fontSize: '8px',
+    width: '8px',
+    height: '8px',
+  },
+
+  // TODO: Remove fontSize once checkbox uses react-icons
+  large: {
+    fontSize: '10px',
+    width: '10px',
+    height: '10px',
+  },
+
+  // TODO: Remove fill once checkbox uses react-icons
+  disabled: theme => ({
+    opacity: 1,
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+    fill: theme.alias.color.neutral.neutralForegroundDisabled,
+  }),
+
+  unchecked: {
+    opacity: 0,
+  },
+
+  // TODO: neutralForegroundInverted change to NeutralForegroundOnBrand once it's added
+  checked: theme => ({
+    opacity: 1,
+    fill: theme.alias.color.neutral.neutralForegroundInverted,
+  }),
+
+  mixed: theme => ({
+    opacity: 1,
+    fill: theme.alias.color.neutral.compoundBrandForeground1,
+    ':hover': {
+      fill: theme.alias.color.neutral.compoundBrandForeground1Hover,
+    },
+    ':active': {
+      fill: theme.alias.color.neutral.compoundBrandForeground1Pressed,
+    },
+  }),
+});
+
+const useLabelStyles = makeStyles({
+  label: {
+    userSelect: 'none',
+    cursor: 'pointer',
+  },
+
+  disabled: theme => ({
+    color: theme.alias.color.neutral.neutralForegroundDisabled,
+    cursor: 'default',
+  }),
+
+  unchecked: theme => ({
+    color: theme.alias.color.neutral.neutralForeground3,
+    ':hover': {
+      color: theme.alias.color.neutral.neutralForeground2,
+    },
+    ':active': {
+      color: theme.alias.color.neutral.neutralForeground1,
+    },
```

Comment:
Out of curiosity, do these hover/active states apply to hovering over _either_ the checkbox _or_ the label? The label should get its hover colors when hovering over the checkbox indicator; and vice versa.

You might need to refactor these styles to use the hover/active states of the root element instead of the label and indicator slots separately.

(Feel free to address this comment in a future PR. It doesn't need to block this PR)